### PR TITLE
Bump Maven from 3.8.7 to 3.9.1 and Mvnd from 0.9.0 to 1.0-m6

### PR DESCRIPTION
**Pull request by bowenliang123**
### _Why are the changes needed?_

- bump Maven from 3.8.7 to 3.9.1, 3.9.1 fixed the performance issue [MNG-7677](https://issues.apache.org/jira/browse/MNG-7677) in 3.9.0, release notes: https://maven.apache.org/docs/3.9.1/release-notes.html
-  Mvnd from 0.9.0 to 1.0-m6 (with embedded maven 3.9.1), release notes: https://github.com/apache/maven-mvnd/releases/tag/1.0-m6

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.readthedocs.io/en/master/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -109,8 +109,9 @@
 
     <properties>
         <java.version>1.8</java.version>
-        <maven.version>3.8.7</maven.version>
-        <mvnd.version>0.9.0</mvnd.version>
+        <maven.version>3.9.1</maven.version>
+        <mvnd.version>1.0-m6</mvnd.version>
+        <mvnd.maven.short.version>39</mvnd.maven.short.version>
```
Comment (pan3793) [inline pom.xml:114]: we don't need it, let's evaluate it in shell script, using `maven.version`

File: build/mvnd [hunk 2]
```diff
@@ -95,6 +95,7 @@ function get_os_arch() {
 # install mvnd under the build/ folder if needed.
 function install_mvnd() {
   local MVND_VERSION=$(grep "<mvnd.version>" "${_DIR}/../pom.xml" | head -n1 | awk -F '[<>]' '{print $3}')
+  local MVND_MVN_SHORT_VERSION=$(grep "<mvnd.maven.short.version>" "${_DIR}/../pom.xml" | head -n1 | awk -F '[<>]' '{print $3}')
```
Comment (pan3793) [inline build/mvnd:98]: Change and move it to the next line
```suggestion
  local MVND_MVN_SHORT_VERSION=$(echo MVN_VERSION | awk -F . '{print $1$2}')
```

File: pom.xml (hunk 1 above)
Comment (bowenliang123) [reply pom.xml:114]: Thanks for the code suggestion. Applied.

**Review by pan3793 — APPROVED** [review]

**Comment by bowenliang123** [comment]
Thanks, merged to master.

**Comment by cxzl25** [comment]
Due to the upgrade of the version of maven, the implementation of http was switched from wagon to aether, and the connection timeout of the two was inconsistent, which caused CI to often fail.

Should we revert this modification first to make CI stable?

### wagon 60s
https://github.com/apache/maven-wagon/blob/master/wagon-provider-api/src/main/java/org/apache/maven/wagon/Wagon.java#L44
### aether 10s
https://github.com/apache/maven-resolver/blob/master/maven-resolver-api/src/main/java/org/eclipse/aether/ConfigurationProperties.java#L93

**Comment by bowenliang123** [comment]
Hi @cxzl25, thanks for the investigation here. The same conclusion and a possible fix to increase connection timeout go to #4692.
I am open to revert this upgrade, but it is more important to figure out how possible way to config the timeout for Maven Resolver as it has been set as default since 3.9.x and 4.x .

**Pull request closed without merging**